commit 1c346dde37a7be9f2ec85a0b41aeea44944d070b
Author: Bachir Soussi Chiadmi <bachir@g-u-i.me>
Date:   Tue Sep 29 15:46:34 2015 +0200

    integrated placing plugin code on at_navigation plugin to be able to sort items

diff --git a/plugins/at_navigation.php b/plugins/at_navigation.php
index c32d79a..59dbc3a 100644
--- a/plugins/at_navigation.php
+++ b/plugins/at_navigation.php
@@ -17,9 +17,36 @@ class AT_Navigation {
 	##
 	# HOOKS
 	##
-
+
+	public function before_read_file_meta(&$headers)
+ 	{
+     	$headers['placing'] = 'Placing';
+ 	}
+
+ 	public function get_page_data(&$data, $page_meta)
+ 	{
+     	$data['placing'] = isset($page_meta['placing']) ? intval($page_meta['placing']) : 0;
+ 	}
+
 	public function get_pages(&$pages, &$current_page, &$prev_page, &$next_page)
 	{
+		global $config;
+		if ($config['pages_order_by'] = 'placing') {
+				$sorted_pages = array();
+
+				$amountDigits = strlen($this->getHighestPlacing($pages));
+
+				$placing_id = 0;
+				foreach ($pages as $page) {
+						$sorted_pages[ $this->formatAmountDigits($page['placing'], $amountDigits) . '-' . $placing_id ] = $page;
+						$placing_id++;
+				}
+
+				ksort($sorted_pages);
+				$pages = $sorted_pages;
+		}
+
+
 		$navigation = array();

 		foreach ($pages as $page)
@@ -30,9 +57,10 @@ class AT_Navigation {
 				$navigation = array_merge_recursive($navigation, $this->at_recursive($_split, $page, $current_page));
 			}
 		}
-
-		array_multisort($navigation);
-		$this->navigation = $navigation;
+
+  	ksort($navigation);
+  	$this->navigation = $navigation;
+
 	}

 	public function config_loaded(&$settings)
@@ -62,7 +90,38 @@ class AT_Navigation {
 	##
 	# HELPER
 	##
-
+
+	/**
+	 * @param array $pages
+	 * @return int
+	 */
+	private function getHighestPlacing(array $pages)
+	{
+			$highest = 0;
+			foreach($pages as $page) {
+					$placing = intval($page['placing']);
+					if( $placing > $highest ) {
+							$highest = $placing;
+					}
+			}
+
+			return $highest;
+	}
+
+	/**
+	 * @param int|string $number
+	 * @param int $digits
+	 * @return string
+	 */
+	private function formatAmountDigits($number, $digits)
+	{
+			while( strlen($number) < $digits ) {
+					$number = '0' . $number;
+			}
+
+			return $number;
+	}
+
 	private function at_build_navigation($navigation = array(), $start = false)
 	{
 		$id = $start ? $this->settings['at_navigation']['id'] : '';
